src/main/scala/com/tapad/docker/DockerComposePlugin.scala

```diff
@@ -154,6 +163,36 @@ class DockerComposePluginLocal extends AutoPlugin with ComposeFile with DockerCo
     }
   }
 
+  def restartRunningInstance(state: State, args: Seq[String]): State = {
+    try {
+      restartInstancePrecheck(state, args)
+      val newState = stopRunningInstances(state, args)
+      launchInstanceWithLatestChanges(newState, args)
+    } catch {
+      case ex: IllegalArgumentException =>
+        printError(ex.getMessage)
+        state
+      case ex: ComposeFileFormatException =>
+        printError(ex.getMessage)
+        state
+    }
+  }
+
+  def restartInstancePrecheck(state: State, args: Seq[String]): Unit = {
```

Comment:
Any function that gets retrieves the state and uses it should return it and the new state should be used by the calling function.